src/Group.php

```diff
@@ -119,48 +119,63 @@ public function cleanDBonPurge()
         Rule::cleanForItemCriteria($this, '_groups_id%');
     }
 
-
     public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0)
     {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
 
-        if (!$withtemplate && self::canView()) {
+        if ($withtemplate || !self::canView()) {
+            return '';
+        }
+
+        $ong = [];
+        if ($item instanceof Group) {
             $nb = 0;
-            switch ($item->getType()) {
-                case 'Group':
-                    /** @var Group $item */
-                    $ong = [];
-                    if ($_SESSION['glpishow_count_on_tabs']) {
-                        $nb = countElementsInTable(
-                            $this->getTable(),
-                            ['groups_id' => $item->getID()]
-                        );
-                    }
-                    $ong[4] = self::createTabEntry(__('Child groups'), $nb);
+            if ($_SESSION['glpishow_count_on_tabs']) {
+                $nb = countElementsInTable($this->getTable(), ['groups_id' => $item->getID()]);
+            }
+            $ong[4] = self::createTabEntry(__('Child groups'), $nb);
+
+            if ($_SESSION['glpishow_count_on_tabs']) {
+                $tree = Session::getSavedOption(__CLASS__, 'tree', 0);
+                $user = Session::getSavedOption(__CLASS__, 'user', 0);
+                $type = Session::getSavedOption(__CLASS__, 'onlytype', '');
```

Comment:
I don't think the tab counter should reflect the filters using in the tab. Any other sublist that has filters doesn't work that way.

Reply:
Agree